pkg/typesystem/typesystem.go

```diff
@@ -561,6 +568,70 @@ func (t *TypeSystem) IsDirectlyRelated(target *openfgav1.RelationReference, sour
 	return false, nil
 }
 
+func (t *TypeSystem) TTUCanFastPathLevel(objectType, relation, userType string, ttu *openfgav1.TupleToUserset, level int) bool {
+	if t.authzWeightedGraph == nil {
+		return false
+	}
+	objRel := tuple.ToObjectRelationString(objectType, relation)
+	tuplesetRelationKey := tuple.ToObjectRelationString(objectType, ttu.GetTupleset().GetRelation())
+	computedRelation := ttu.GetComputedUserset().GetRelation()
+	node, ok := t.authzWeightedGraph.GetNodeByID(objRel)
+	if !ok {
+		return false
+	}
+
+	w, ok := node.GetWeight(userType)
+	if !ok {
+		return false
+	}
+
+	if w == level && len(node.GetWildcards()) == 0 {
+		return true
+	}
```

Comment:
This block puzzles me. If we don't enter it, it could mean that the weight of the node is >2, in that case, we can still return true? But i thought we _only_ wanted nodes of weight 2? 

Also, in my mind, I would have written


```
len(node.GetWildcards()) > 0 {
		return false
	}
```

regardless of the weight?

We can jump on a Zoom if it's too complex to explain here

Reply:
it took me a while to digest this bit. @elbuo8 can we leave a comment in the code? something like:

> `node.weight(userType)` is only the maximum weight to get to "userType". if the weight is exactly 2, we exit early. if the weight is >2, it is possible that the node has multiple paths to arrive to "userType", but the path that we are exploring right now actually has weight <= 2, in which case we want to return `true`.